aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotTLSSocketFactory.java

```diff
@@ -105,6 +109,9 @@ private Socket enableTLSOnSocket(Socket socket) {
             ((SSLSocket) socket).setEnabledProtocols(new String[] {
                 "TLSv1.2"
             });
+            if (Conscrypt.isConscrypt((SSLSocket) socket)) {
```

Comment:
Only set the ALPN extension if interacting with the server on port 443.

Reply:
Discussed offline. We want to single Provider for all the ports and IoT supports ALPN for both 443 and 8883